gps-rtk-serial-no-network/gps_rtk_serial_no_network.go

```diff
@@ -0,0 +1,413 @@
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/utils"
+)
+
+var Model = resource.NewModel("viam-labs", "movement-sensor", "gps-rtk-serial-no-network")
+
+const serialStr = "serial"
+
+type Config struct {
+	SerialNMEAPath           string `json:"serial_NMEA_path"`
+	SerialNMEABaudRate       int    `json:"serial__NMEA_baud_rate,omitempty"`
+	SerialCorrectionPath     string `json:"serial_correction_path"`
+	SerialCorrectionBaudRate int    `json:"serial_correction_baud_rate"`
+}
+
+// ValidateSerial ensures all parts of the config are valid.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	var deps []string
+	if cfg.SerialNMEAPath == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "serial_NMEA_path")
+	}
+	if cfg.SerialCorrectionPath == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "serial_correction_path")
+	}
+	return deps, nil
+}
+
+func init() {
+	resource.RegisterComponent(
+		movementsensor.API,
+		Model,
+		resource.Registration[movementsensor.MovementSensor, *Config]{
+			Constructor: func(
+				ctx context.Context,
+				deps resource.Dependencies,
+				conf resource.Config,
+				logger golog.Logger,
+			) (movementsensor.MovementSensor, error) {
+				newConf, err := resource.NativeConfig[*Config](conf)
+				if err != nil {
+					return nil, err
+				}
+				return newRTKSerialNoNetwork(ctx, deps, conf.ResourceName(), newConf, logger)
+			},
+		})
+}
+
+// A RTKSerialNoNetwork is a MovementSensor model that can intake RTK correction data from a serial path.
+type RTKSerialNoNetwork struct {
+	resource.Named
+	resource.AlwaysRebuild
+	logger     golog.Logger
+	cancelCtx  context.Context
+	cancelFunc func()
+
+	activeBackgroundWorkers sync.WaitGroup
+
+	err          movementsensor.LastError
+	lastposition movementsensor.LastPosition
+
+	Nmeamovementsensor gpsnmea.NmeaMovementSensor
+	CorrectionWriter   io.ReadWriteCloser
+	CorrectionReader   io.ReadCloser
+	correctionReaderMu sync.Mutex
+
+	writePath     string
+	writeBaudRate int
+
+	readPath     string
+	readBaudRate int
+}
+
+func newRTKSerialNoNetwork(
+	ctx context.Context,
+	deps resource.Dependencies,
+	name resource.Name,
+	newConf *Config,
+	logger golog.Logger,
+) (movementsensor.MovementSensor, error) {
+
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	g := &RTKSerialNoNetwork{
+		Named:        name.AsNamed(),
+		cancelCtx:    cancelCtx,
+		cancelFunc:   cancelFunc,
+		logger:       logger,
+		err:          movementsensor.NewLastError(1, 1),
+		lastposition: movementsensor.NewLastPosition(),
+	}
+
+	nmeaConf := &gpsnmea.Config{
+		ConnectionType: serialStr,
+		DisableNMEA:    false,
+	}
+
+	// Init NMEAMovementSensor
+	nmeaConf.SerialConfig = &gpsnmea.SerialConfig{SerialPath: newConf.SerialNMEAPath, SerialBaudRate: newConf.SerialNMEABaudRate}
+	var err error
+	g.Nmeamovementsensor, err = gpsnmea.NewSerialGPSNMEA(ctx, name, nmeaConf, logger)
```

Comment:
is there a way to not use the Nmemovement sensor type? Or are there in rdk changes that need to happen first?

Reply:
Not sure if I understand your question correctly, but I could call the `newNMEAGPS` constructor and set the connection type to serial and then inside that function `NewSerialGPSNMEA` would be called.

Reply:
We can do this in a follow up pr, but imaging that the `NmeaMovementSensor` is no longer exported from the in rdk package at some point in the future. Is there a design where this driver can self-contain all the information needed to return all the API related data. 

Maybe a really big lift, so we don't need to do it now.

Reply:
oh yes I can move the NMEA constructors and parsers to this module, will do that in another PR.